src/builder.rs

```diff
@@ -124,6 +137,47 @@ impl StatsdBuilder {
         self
     }
 
+    /// Use a custom `MetricSink`.
+    ///
+    /// This method supersedes all other settings for metrics output, including the hostname and
+    /// port specified in [`StatsdBuilder::from`] and values passed to the `with_queue_size`,
+    /// `with_buffer_size`, and `with_client_udp_host` methods. The specified `sink` is used
+    /// instead.
+    ///
+    /// (When this method is not called, the builder creates a default sink using those settings,
+    /// [`cadence::QueuingMetricSink`], and [`cadence::UdpMetricSink`].)
+    ///
+    /// # Examples
+    ///
+    /// This code replaces the ordinary UDP sink with output to a Unix socket.
+    ///
+    /// ```no_run
+    /// use metrics_exporter_statsd::StatsdBuilder;
+    /// use cadence::BufferedUnixMetricSink;
+    /// use std::os::unix::net::UnixDatagram;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error + Sync + Send>> {
+    /// let path = "/path/to/my/metrics/socket";
+    /// let socket = UnixDatagram::bind(path)?;
+    /// let sink = BufferedUnixMetricSink::from(path, socket);
+    ///
+    /// let recorder = StatsdBuilder::from("", 0)
+    ///     .with_sink(sink)
+    ///     .build(Some("my_app"))?;
+    /// metrics::set_boxed_recorder(Box::new(recorder));
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn with_sink<T>(mut self, sink: T) -> Self
```

Comment:
Silently overriding the other sink configs is a pretty big surprise. Maybe this could be implemented here in a way that bypasses the builder type?

For example: `fn StatsdBuilder::from_sink<T>(sink: T) -> StatsdRecorder`

That being said, I'm not sad about the public `StatsdRecorder` usage either:

```rust
let sink = build_sink();
let statsd_client = StatsdClient::from_sink("prefix", sink);
let recorder = StatsdRecorder::new(statsd_client, HistogramType::Histogram);
metrics::set_boxed_recorder(Box::new(recorder))
  .expect("metrics recorder should be uninitialized");
```

Reply:
I feel a little sheepish about this, but I sketched it out in the way you suggest, and I think this is probably the least awkward way. In any case we can improve on it later. Thanks for your patience.